channels/api.py

```diff
@@ -350,6 +354,63 @@ def sync_comment_model(*, channel_name, post_id, comment_id, parent_id):
         )[0]
 
 
+def sync_channel_subscription_model(channel_name, user):
+    """
+    Create or update channel subscription for a user
+
+    Args:
+        channel_name (str): The name of the channel
+        user (django.contrib.models.auth.User): The user
+
+    Returns:
+        ChannelSubscription: the channel user role object
+    """
+    with transaction.atomic():
+        channel = sync_channel_model(channel_name)
```

Comment:
Can leave this for now as I mentioned in my other comment, I'll refactor the existing sync_* methods away in my other PR.